pkg/pub_worker/lib/src/analyze.dart

```diff
@@ -152,12 +152,27 @@ Future<void> _analyzePackage(
       await pana.stdin.close();
 
       await Future.wait<void>([
-        pana.stderr.forEach(log.add),
-        pana.stdout.forEach(log.add),
+        pana.stderr
+            .transform(Utf8Decoder(allowMalformed: true))
+            .transform(LineSplitter())
+            .forEach(log.writeln),
+        pana.stdout
+            .transform(Utf8Decoder(allowMalformed: true))
+            .transform(LineSplitter())
+            .forEach(log.writeln),
         pana.exitOrTimeout(_panaTimeout, () {
-          log.writeln('TIMEOUT: pana sending SIGTERM/SIGKILL');
+          log.writeln('TIMEOUT: sending SIGTERM/SIGKILL to pana.');
         }),
-      ]).catchError((e) => const [/* ignore */]);
+      ]).catchError((e, st) {
+        log.writeln('TIMEOUT: sending SIGKILL to pana.');
+        // kill the process if not already killed
+        try {
+          pana.kill(ProcessSignal.sigkill);
+        } finally {
+          // ignore
+        }
+        return const [];
```

Comment:
`Future.wait` won't return until all futures are done.

And `exitOrTimeout` does afaik the right thing:
https://github.com/dart-lang/pub-dev/blob/1f628e218caca0a3fecc9201ac465068c2d1e391/pkg/pub_worker/lib/src/analyze.dart#L302-L326

It even does the graceful shutdown thing.

If for some reason the pipe to stderr should fail (say diskspace is full and `log.writeln` fails), then we'll just hang waiting for the timeout.

Then what is the point of calling `sigkill` again.

I think we should ignore the error here as we did before, I don't see any reason to catch log it.

I wouldn't mind a `.catchError((e, st) => _log....` after `.forEach(log.writeln)` (in both places).
But we could also just live without it.

Reply:
Good points. I've reverted this part.